### PR TITLE
Use CTS type wrappers for test_global_linear_id.

### PR DESCRIPTION
**Pull request by JohnKesapidesARM**
Signed-off-by: John Kesapides <john.kesapides@arm.com>

### CODE REVIEW

File: test_conformance/basic/test_global_linear_id.cpp [hunk 1]
```diff
@@ -22,105 +22,93 @@
...
+    for (int i = 0; i < ARRAY_SIZE(program); i++)
+    {
+        err = create_single_kernel_helper(context, &program[i], &kernel[i], 1,
+                                          &kernel_code[i], kernel_names[i]);
+        test_error(err, "create_single_kernel_helper failed");
+
+        err = clSetKernelArg(kernel[i], 0, sizeof streams, &streams);
+        test_error(err, "clSetKernelArgs failed.");
+
+        err = clEnqueueNDRangeKernel(queue, kernel[i], i + 1, nullptr, threads,
```
Comment (EwanC) [inline test_conformance/basic/test_global_linear_id.cpp:101]: In the old code we had

* line 100 - `err = clEnqueueNDRangeKernel(queue, kernel[1], 2,...`
* line 109 - `err = clEnqueueNDRangeKernel(queue, kernel[1], 1,..`

So they were both using `kernel[1]`, and `kernel[0]` never seemed to be enqueued, was this a bug?

Comment (JohnKesapidesARM) [reply test_conformance/basic/test_global_linear_id.cpp:101]: yeap indeed it was ;).

**Review by EwanC — APPROVED** [review]

File: test_conformance/basic/test_global_linear_id.cpp (hunk 1 above)
Comment (kpet) [reply test_conformance/basic/test_global_linear_id.cpp:101]: Hmm, I think the test as refactored is broken. The first kernel will only write to the first `num_elements` value of the buffer (that is not pre-initialised) and yet the verification code will always check `num_elements * num_elements` values. Am I missing something?

File: test_conformance/basic/test_global_linear_id.cpp [hunk 2]
```diff
@@ -22,105 +22,93 @@
...
+                                     nullptr, 0, nullptr, nullptr);
+        test_error(err, "clEnqueueNDRangeKernel failed.");
+
+        err = clEnqueueReadBuffer(queue, streams, CL_TRUE, 0,
+                                  length * sizeof(int), output.data(), 0,
+                                  nullptr, nullptr);
+        test_error(err, "clEnqueueReadBuffer failed.");
+
+        err = verify_global_linear_id(output, length);
```
Comment (kpet) [inline test_conformance/basic/test_global_linear_id.cpp:110]: Verification probably fails in the first iteration of the loop (see comment above) but the status isn't checked and in the second iteration the error gets overwritten.

**Review by kpet — CHANGES_REQUESTED** [review]

File: test_conformance/basic/test_global_linear_id.cpp (hunk 1 above)
Comment (JohnKesapidesARM) [reply test_conformance/basic/test_global_linear_id.cpp:101]: hmm, looks like you are right. I replaced one bug with another ;)

File: test_conformance/basic/test_global_linear_id.cpp [hunk 3]
```diff
@@ -22,105 +22,93 @@
...
-    clReleaseProgram(program[0]);
-    clReleaseProgram(program[1]);
-    free(output_ptr);
+
+    num_elements = static_cast<int>(sqrt(static_cast<float>(num_elements)));
+    int length = 1;
+    size_t threads[] = { static_cast<size_t>(num_elements),
+                         static_cast<size_t>(num_elements) };
+
+    for (int i = 0; i < ARRAY_SIZE(program) && !err; i++)
```
Comment (EwanC) [inline test_conformance/basic/test_global_linear_id.cpp:84]: we are checking `&& !err` here, but `int err;` on line 77 isn't getting initialized before the first loop iteration. Think line 77 should be `int err = CL_SUCCESS;`

Comment (JohnKesapidesARM) [reply test_conformance/basic/test_global_linear_id.cpp:84]: yeap you are right, thanks for spotting that. I'll fix asap

**Review by EwanC — APPROVED** [review]

**Review by lakshmih — APPROVED** [review]

**Review by bashbaug — APPROVED** [review]

**Review by kpet — APPROVED** [review]

**Pull request merged**